src/api/records/record.controller.js

```diff
@@ -889,9 +892,9 @@ export async function validate(req, res) {
         }
       }
     })
-    .on('end', function() {
-      console.log(`Registros encontrados: ${amount}`);
-      console.log(`Errores encontrados: ${errors}`);
-      res.send(`Errores encontrados: ${errors}`);
+    .on('end', async function() {
+      await console.log(`Registros encontrados: ${amount}`);
+      await console.log(`Errores encontrados: ${errors}`);
+      await res.send(`Errores encontrados: ${errors}`);
```

Comment:
I don't understand what is the `async` - `await` for

Reply:
I noticed that this callback is not executed properly. Sometimes the response is returned with a wrong value. I think is due to  this callback is not asynchronous unlike the first one (`data`).

If you consider i can let this callback as it was before o we can discuss it tomorrow.

Reply:
`await` is used to avoid the `then-catch` on a promise, in this case it doesn't make sense since neither `console.log` or `res.send` are promises. The `on('data')` function has inside the `record.validate()` call which returns a [promise](https://mongoosejs.com/docs/api/model.html#model_Model.validate).

If the endpoint is returning wrong values is for another reason.

Reply:
`Async` and `await` is removed from `on('end')`